zproject/backends.py

```diff
@@ -319,6 +321,19 @@ def ldap_to_django_username(self, username: str) -> str:
         return username
 
 class ZulipLDAPAuthBackend(ZulipLDAPAuthBackendBase):
+    fake_ldap_auth_enabled = getattr(settings, 'FAKE_LDAP_AUTH_ENABLED', False)
+
+    if settings.DEVELOPMENT and fake_ldap_auth_enabled:
+        mode = getattr(settings, 'FAKE_LDAP_MODE', 'a')
+        extra_users = getattr(settings, 'FAKE_LDAP_EXTRA_USERS', 0)
+
+        ldap_patcher = mock.patch('django_auth_ldap.config.ldap.initialize')
+        mock_initialize = ldap_patcher.start()
+        mock_ldap = MockLDAP()
+        mock_initialize.return_value = mock_ldap
+
+        mock_ldap.directory = generate_dev_ldap_dir(mode, extra_users)
```

Comment:
Should this code live in an `__init__` function?

otherwise all these variables end up exported via `self` just like `fake_ldap_auth_enabled`